### PR TITLE
fix failing thermomechanics test

### PR DESCRIPTION
**Pull request by samuelpmishLLNL**
the relevant fix is the small change to solid_mechanics.hpp, where our extrapolated initial guess for the Newton solver wasn't respecting the essential boundary conditions. Was there a related issue for this failing test?

I'm also adding the reproducer file from #940 to facilitate future debugging of that issue.

### CODE REVIEW

**Comment by btalamini** [comment]
This is a great catch. It's troubling that none of our other tests were sensitive to this. I guess that our patch tests were passing because they were taking more than 1 Newton iteration, and we recovered the essential boundary conditions on the second and subsequent Newton iterations.
I think we would catch this if we disallow Newton iterations on the patch tests. The patch solutions are linear, so the warm start should hit the exact solution without any subsequent Newton iterations.

**Review by btalamini — APPROVED** [review]

File: src/serac/physics/tests/beam_bending.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  axom::sidre::DataStore datastore;
+  serac::StateManager::initialize(datastore, "beam_bending_data");
+
+  // Construct the appropriate dimension mesh and give it to the data store
+  std::string filename = SERAC_REPO_DIR "/data/meshes/beam-quad.mesh";
+
+  auto mesh = mesh::refineAndDistribute(buildMeshFromFile(filename), 0, 0);
+  serac::StateManager::setMesh(std::move(mesh));
+
+  // _solver_params_start
```
Comment (jamiebramwell) [inline src/serac/physics/tests/beam_bending.cpp]: I believe these are copied from the other example that I pull into the docs page. They should be deleted here.

**Review by jamiebramwell — APPROVED** [review]
Thank you for figuring this out!
